src/main/java/core/basesyntax/strategy/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,10 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.storage.Storage;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void apply(String fruit, int quantity) {
+        Storage.fruits.put(fruit, Storage.fruits.get(fruit) + quantity);
```

Comment:
There is a logical error here. For a purchase operation, the quantity should be subtracted from the existing quantity in storage, not added. Consider changing the operation to `Storage.fruits.put(fruit, Storage.fruits.get(fruit) - quantity);` to reflect a purchase.